sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ManagementChannel.java

```diff
@@ -97,6 +99,32 @@
                 requestMessage.setBody(new AmqpValue(Collections.singletonMap(ManagementConstants.SEQUENCE_NUMBERS,
                     new Long[]{sequenceNumber})));
 
+                TransactionalState transactionalState = null;
+                if (transactionContext != null && transactionContext.getTransactionId() != null) {
+                    transactionalState = new TransactionalState();
+                    transactionalState.setTxnId(new Binary(transactionContext.getTransactionId().array()));
+                }
+
+                return sendWithVerify(channel, requestMessage, transactionalState);
+            })).then();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Mono<Void> cancelScheduledMessages(Iterable<Long> sequenceNumbers, String associatedLinkName) {
+        return isAuthorized(ManagementConstants.OPERATION_CANCEL_SCHEDULED_MESSAGE)
```

Comment:
Why duplicate logic, the logic in cancel and schedule with iterable and single item are exactly the same.